apps/python/scdbstrip.py

```diff
@@ -720,40 +757,43 @@ def clean(self):
             else:
                 timestamp = self._datetime
 
-            if not self._invertMode:
-                output.write(
-                    "[INFO] Keep objects after %s UTC\n"
-                    % timestamp.toString("%Y-%m-%d %H:%M:%S")
-                )
-            else:
-                output.write(
-                    "[INFO] Keep objects before %s UTC\n"
-                    % timestamp.toString("%Y-%m-%d %H:%M:%S")
-                )
+            def timeRangeSelection(what):
+                if self._daterange:
+                    if self._invertMode:
+                        formatstr = "( %s < '%s' or %s > '%s' )"
+                    else:
+                        formatstr = "( %s >= '%s' and %s <= '%s' )"
```

Comment:
For consistency with the `date` option, wouldn't it be better to use `formatstr = "( %s >= '%s' and %s < '%s' )"`? The convention is to never include the upper time in the search range to get distinct intervals for, e.g. 10:00-11:00 and 11:00-12:00.

Reply:
sounds good, let me fix this

Reply:
No wait, it feels strange to not include the upper time, because the `invert` option wouldn't include that time either and it feels wrong that a particular point in time remains unhandled.

Reply:
Then we must include `timestamp` as well ( https://github.com/SeisComP/main/blob/ed368aff7a2421629106552d37685e6545b234dc/apps/python/scdbstrip.py#L773) which I don't like.

It might feel strange, but seedlink works like that and the record stream implementations. They use `< endTime` to avoid duplicate records for adjacent intervals. This convention feels like starting to count from zero rather from one. ;)

Reply:
ok then, if you are sure about that then I am ok